### PR TITLE
[stable/mongodb-replicaset] Refactor tests

### PR DESCRIPTION
**Pull request by unguiculus**
* Extract test script to separate file which makes it easier to develop and debug 
* Apply fixed suggested by Shellcheck 
* Consider configured number of replicas 
* Apply standard labels to test resources

### CODE REVIEW

**Comment by unguiculus** [comment]
/cc @foxish 

Based on #1619 which should be merged first.

File: stable/mongodb-replicaset/mongodb-up-test.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+    pod_name() {
+        local full_name="${FULL_NAME?Environment variable FULL_NAME not set}"
+        local index="$1"
+        echo "$full_name-$index.$full_name.default.svc.cluster.local"
```
Comment (foxish) [inline stable/mongodb-replicaset/mongodb-up-test.sh]: This assumes `default` namespace, and a certain cluster domain: `svc.cluster.local`, which recently tripped people up.
It is a test, but I'm not sure that we should make this assumption.

Comment (foxish) [reply stable/mongodb-replicaset/mongodb-up-test.sh]: I know it is no worse than it was before, but maybe we should revisit this now.

**Comment by unguiculus** [comment]
I no longer fully qualify the pods in the test, which is ok because the test runs in the same namespace.

**Comment by foxish** [comment]
Thanks @unguiculus

**Pull request merged**